src/c++/vdo/base/dm-vdo-target.c

```diff
@@ -3058,7 +3078,8 @@ static int vdo_preresume(struct dm_target *ti)
 
 	vdo_register_thread_device_id(&instance_thread, &vdo->instance);
 	result = vdo_preresume_registered(ti, vdo);
-	if ((result == VDO_PARAMETER_MISMATCH) || (result == VDO_INVALID_ADMIN_STATE))
+	if ((result == VDO_PARAMETER_MISMATCH) || (result == VDO_INVALID_ADMIN_STATE) ||
+	    (result == VDO_UNSUPPORTED_VERSION))
 		result = -EINVAL;
```

Comment:
As noted above, you don't need to change this line; just return the value you really want from vdo_preresume_registere().

Reply:
I don't agree. Because we have a status code associated with the failure, I think it makes the most sense to add it here to the list translated to -EINVAL. That way any status code that results in -EINVAL being returned is listed in a single location. It maintains consistency and code clarity, in my opinion.

Reply:
Yeah, okay, that makes sense. This spelling is fine.